expression/column.go

```diff
@@ -73,7 +73,15 @@ func (col *CorrelatedColumn) EvalString(ctx sessionctx.Context, row chunk.Row) (
 	}
 	res, err := col.Data.ToString()
 	resLen := len([]rune(res))
-	if resLen < col.RetType.Flen && ctx.GetSessionVars().StmtCtx.PadCharToFullLength {
+	if col.GetType().Tp != mysql.TypeString {
+		return res, err != nil, errors.Trace(err)
+	}
+	// If sql_mode 'pad_char_to_full_length' is no set, the right trailing spaces should be removed for type 'CHAR'.
```

Comment:
how about: `If the 'pad_char_to_full_length' sql_mode is not set, the right padding spaces should be removed for type 'CHAR'.`

Reply:
I think `If 'pad_char_to_full_length' is set in 'sql_mode' system variable, .....` would be better, that's more consistent with https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html